<a href="https://colab.research.google.com/github/Nunticha24/DWDM21/blob/main/Project_Dinosaur.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

#Present Project กลุ่ม

https://drive.google.com/drive/u/0/folders/1G9XaOQ1IExr0m-UwWypqGPtGzi5HE-YL

In [ ]:
import pandas as pd
import os

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
path = '/content/drive/My Drive/Project_Dinosaur'

#data covid

##ตารางข้อมูล covid ปี2020

* No รหัสผู้ป่วย
* Age อายุของผู้ป่วย
* Sex เพศผู้ป่วย
* Nationality สัญชาติ
* Province of isolation จังหวัดที่รักษาตัว
* Notification Date วันที่กระทรวงได้รับรายงาน
* Announce Date วันที่แถลงข่าว
* Province of onset จังหวัดที่เริ่มพบว่าตัวเองมีอาการป่วย
* District of onset อำเภอที่เริ่มพบว่าตัวเองมีอาการป่วย
* risk สถานที่

In [ ]:
data_covid2020 =  pd.read_csv(os.path.join(path,'confirmed-cases.csv'))
print(data_covid2020.shape)
data_covid2020.head()

(839771, 11)


/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


No. announce_date Notified date   sex   age Unit nationality  \
0    1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1    2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2    3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3    4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4    5     24/1/2020           NaN  หญิง  66.0   ปี       China   

  province_of_isolation                              risk province_of_onset  \
0         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4               นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   

  district_of_onset  
0               NaN  
1               NaN  
2             เมือง  
3               NaN  
4               NaN

##ตารางข้อมูล covid ปี2021(1)

In [ ]:
data_covid2021_1 =  pd.read_csv(os.path.join(path,'confirmed-cases-since-120864.csv'))
print(data_covid2021_1.shape)
data_covid2021_1.head()

(1042168, 11)


No. announce_date Notified date   sex   age Unit nationality  \
0  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
1  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
2  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
3  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
4  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0              เชียงราย                                        อื่นๆ   
1              เชียงราย                                        อื่นๆ   
2              เชียงราย                                        อื่นๆ   
3              เชียงราย                                        อื่นๆ   
4              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN

##ตารางข้อมูล covid ปี2021(2)

In [ ]:
data_covid2021_2 =  pd.read_csv(os.path.join(path,'confirmed-cases-since-271064.csv'))
print(data_covid2021_2.shape)
data_covid2021_2.head()

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:2882: DtypeWarning: Columns (4) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


(1010459, 12)


No. announce_date Notified date   sex   age Unit nationality  \
0  1859158    26/10/2021    25/10/2021   ชาย  20.0   ปี    Thailand   
1  1859159    26/10/2021    25/10/2021  หญิง  42.0   ปี    Thailand   
2  1859160    26/10/2021    25/10/2021  หญิง  33.0   ปี    Thailand   
3  1859161    26/10/2021    25/10/2021  หญิง  52.0   ปี    Thailand   
4  1859162    26/10/2021    25/10/2021  หญิง  84.0   ปี    Thailand   

  province_of_isolation                                           risk  \
0                 ลำปาง                                          อื่นๆ   
1                 ลำปาง                                          อื่นๆ   
2                 ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
3                 ลำปาง  ไปสถานที่ชุมชน เช่น ตลาดนัด สถานที่ท่องเที่ยว   
4                 ลำปาง    สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  Unnamed: 11  
0             ลำปาง        เมืองลำปาง          NaN  
1             ลำปาง        เมืองลำปาง          NaN  
2             ลำปาง          ห้างฉัตร          NaN  
3             ลำปาง          ห้างฉัตร          NaN  
4             ลำปาง        เมืองลำปาง          NaN

##ตารางข้อมูล covid ปี2022

In [ ]:
data_covid2022 =  pd.read_csv(os.path.join(path,'confirmed-cases-since-280265.csv'))
print(data_covid2022.shape)
data_covid2022.head()

(433553, 11)


No. announce_date Notified date   sex   age Unit nationality  \
0  2869617     28/2/2022     27/2/2022   ชาย  41.0   ปี    Thailand   
1  2869618     28/2/2022     27/2/2022  หญิง  45.0   ปี    Thailand   
2  2869619     28/2/2022     27/2/2022  หญิง  40.0   ปี    Thailand   
3  2869620     28/2/2022     27/2/2022   ชาย  44.0   ปี    Thailand   
4  2869621     28/2/2022     27/2/2022   ชาย  11.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
1                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
2                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
3                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
4                 พะเยา  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0             พะเยา        เมืองพะเยา  
1             พะเยา        เมืองพะเยา  
2             พะเยา        เมืองพะเยา  
3             พะเยา        เมืองพะเยา  
4             พะเยา        เมืองพะเยา

###ตรวจสอบว่าข้อมูลซ้ำกันหรือไม่

าค่า min max ของ column NO.

In [ ]:
def min_max(i):
    a = min(i['No.'])
    b = max(i['No.'])
    print(f'min of NO. is {a}')
    print(f'max of NO. is {b}')

ดูค่า min max ของ covid ทั้ง 4 ตาราง

In [ ]:
min_max(data_covid2020)

min of NO. is 1
max of NO. is 839771


In [ ]:
min_max(data_covid2021_1)

min of NO. is 816990
max of NO. is 1859157


In [ ]:
min_max(data_covid2021_2)

min of NO. is 1859158
max of NO. is 2869616


In [ ]:
min_max(data_covid2022)

min of NO. is 2869617
max of NO. is 3303169


เห็นได้ว่าลักษณะของข้อมูลที่ดูจากค่าในคอลัมน์ NO. ของตาราง data_covid2020 และ data_covid2021_1 มีค่าซ้ำกัน

###ดูลักษณะข้อมูลว่าซ้ำกันจริงหรือไม่

In [ ]:
data_covid2020.iloc[816989:,:].head(10)    #ดูข้อมูล data_covid2020 ตั้งแต่แถวที่ 816989 ถึงแถวสุดท้าย

No. announce_date Notified date   sex   age Unit nationality  \
816989  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
816990  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
816991  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
816992  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
816993  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
816994  816995     12/8/2021     11/8/2021   ชาย  39.0   ปี    Thailand   
816995  816996     12/8/2021     11/8/2021   ชาย  45.0   ปี    Thailand   
816996  816997     12/8/2021     11/8/2021  หญิง  32.0   ปี    Thailand   
816997  816998     12/8/2021     11/8/2021  หญิง   7.0   ปี    Thailand   
816998  816999     12/8/2021     11/8/2021   ชาย   8.0   ปี    Thailand   

       province_of_isolation                                         risk  \
816989              เชียงราย                                        อื่นๆ   
816990              เชียงราย                                        อื่นๆ   
816991              เชียงราย                                        อื่นๆ   
816992              เชียงราย                                        อื่นๆ   
816993              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
816994              เชียงราย                                        อื่นๆ   
816995              เชียงราย                                        อื่นๆ   
816996              เชียงราย                                        อื่นๆ   
816997              เชียงราย                                        อื่นๆ   
816998              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

       province_of_onset district_of_onset  
816989               NaN               NaN  
816990               NaN               NaN  
816991               NaN               NaN  
816992               NaN               NaN  
816993               NaN               NaN  
816994               NaN               NaN  
816995               NaN               NaN  
816996               NaN               NaN  
816997               NaN               NaN  
816998               NaN               NaN

In [ ]:
data_covid2021_1.head(10)    #ดูข้อมูล data_covid2021_1

No. announce_date Notified date   sex   age Unit nationality  \
0  816990     12/8/2021     11/8/2021   ชาย   7.0   ปี    Thailand   
1  816991     12/8/2021     11/8/2021   ชาย   1.0   ปี    Thailand   
2  816992     12/8/2021     11/8/2021   ชาย  35.0   ปี    Thailand   
3  816993     12/8/2021     11/8/2021  หญิง  33.0   ปี    Thailand   
4  816994     12/8/2021     11/8/2021  หญิง  14.0   ปี    Thailand   
5  816995     12/8/2021     11/8/2021   ชาย  39.0   ปี    Thailand   
6  816996     12/8/2021     11/8/2021   ชาย  45.0   ปี    Thailand   
7  816997     12/8/2021     11/8/2021  หญิง  32.0   ปี    Thailand   
8  816998     12/8/2021     11/8/2021  หญิง   7.0   ปี    Thailand   
9  816999     12/8/2021     11/8/2021   ชาย   8.0   ปี    Thailand   

  province_of_isolation                                         risk  \
0              เชียงราย                                        อื่นๆ   
1              เชียงราย                                        อื่นๆ   
2              เชียงราย                                        อื่นๆ   
3              เชียงราย                                        อื่นๆ   
4              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   
5              เชียงราย                                        อื่นๆ   
6              เชียงราย                                        อื่นๆ   
7              เชียงราย                                        อื่นๆ   
8              เชียงราย                                        อื่นๆ   
9              เชียงราย  สัมผัสใกล้ชิดกับผู้ป่วยยืนยันรายก่อนหน้านี้   

  province_of_onset district_of_onset  
0               NaN               NaN  
1               NaN               NaN  
2               NaN               NaN  
3               NaN               NaN  
4               NaN               NaN  
5               NaN               NaN  
6               NaN               NaN  
7               NaN               NaN  
8               NaN               NaN  
9               NaN               NaN

จากการดูลักษณะข้อมูล จะเห็นว่าข้อมูลซ้ำกันจริง

จึงทำการลบข้อมูลที่ซ้ำออก โดยเลือกจากตาราง data_covid2020

In [ ]:
data_covid2020_drop = data_covid2020.iloc[:816989,:]
data_covid2020_drop.head()

No. announce_date Notified date   sex   age Unit nationality  \
0    1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1    2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2    3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3    4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4    5     24/1/2020           NaN  หญิง  66.0   ปี       China   

  province_of_isolation                              risk province_of_onset  \
0         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4               นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   

  district_of_onset  
0               NaN  
1               NaN  
2             เมือง  
3               NaN  
4               NaN

In [ ]:
min_max(data_covid2020_drop)   #ดู No. น้อยสุดและมากสุด ใน data_covid2020_drop

min of NO. is 1
max of NO. is 816989


In [ ]:
min_max(data_covid2022) #ดู No. น้อยสุดและมากสุด ใน data_covid2022

min of NO. is 2869617
max of NO. is 3303169


บวกดูจำนวนข้อมูลของทั้ง 4 ตาราง

In [ ]:
data_covid2020_drop.shape[0] + data_covid2021_1.shape[0] + data_covid2021_2.shape[0] + data_covid2022.shape[0]

3303169

จะเห็นว่าจำนวน NO. ที่มากที่สุดของข้อมูล เท่ากับจำนวนข้อมูลของทั้ง 4 ตาราง บวกกัน ซึ่งเท่ากับ 3303169 แสดงว่าข้อมูลแต่ละตารางไม่ได้เก็บซ้ำกันแล้ว

##รวมตาราง data_covid ทั้ง 4 ตาราง

In [ ]:
data_covid = pd.concat([data_covid2020_drop, data_covid2021_1, data_covid2021_2, data_covid2022])
print(data_covid.shape)
data_covid.head()

(3303169, 12)


No. announce_date Notified date   sex   age Unit nationality  \
0    1     12/1/2020           NaN  หญิง  61.0   ปี       China   
1    2     17/1/2020           NaN  หญิง  74.0   ปี       China   
2    3     22/1/2020           NaN  หญิง  73.0   ปี    Thailand   
3    4     22/1/2020           NaN   ชาย  68.0   ปี       China   
4    5     24/1/2020           NaN  หญิง  66.0   ปี       China   

  province_of_isolation                              risk province_of_onset  \
0         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
1         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
2                นครปฐม  คนต่างชาติเดินทางมาจากต่างประเทศ            นครปฐม   
3         กรุงเทพมหานคร  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   
4               นนทบุรี  คนต่างชาติเดินทางมาจากต่างประเทศ     กรุงเทพมหานคร   

  district_of_onset  Unnamed: 11  
0               NaN          NaN  
1               NaN          NaN  
2             เมือง          NaN  
3               NaN          NaN  
4               NaN          NaN

###เลือกเอา column ที่ต้องการ

In [ ]:
covid = data_covid[['announce_date','province_of_isolation']]
print(covid.shape)
covid.head()

(3303169, 2)


announce_date province_of_isolation
0     12/1/2020         กรุงเทพมหานคร
1     17/1/2020         กรุงเทพมหานคร
2     22/1/2020                นครปฐม
3     22/1/2020         กรุงเทพมหานคร
4     24/1/2020               นนทบุรี

###Clean data_covid

In [ ]:
Covid.isnull().any()

announce_date            False
province_of_isolation     True
dtype: bool